cohort-evaluator-spark/src/test/java/com/ibm/cohort/cql/spark/SparkCqlEvaluatorTest.java

```diff
@@ -125,7 +129,132 @@ public void testAllTypesEvaluationSuccess() throws Exception {
+                "-o", "Context1Id=" + context1IdFile.toURI().toString(),
+                "-o", "Context2Id=" + context2IdFile.toURI().toString(),
+                "--overwrite-output-for-contexts"
+        };
+
+        SparkCqlEvaluator.main(args);
+        
+        
+        validateOutput(
+                context1IdFile.toURI().toString(),
+                Arrays.asList(
+                        RowFactory.create(0, null, null, null, false),
+                        RowFactory.create(1, null, null, null, false),
+                        RowFactory.create(2, 33, "string1", new BigDecimal(9.989), true),
+                        RowFactory.create(3, 22, "string2", new BigDecimal(-2.816), true),
+                        RowFactory.create(4, 22, "string1", new BigDecimal(-4.926), true)
+                ),
+                new StructType()
+                        .add("id", DataTypes.IntegerType, false)
+                        .add("Context1Id.define_integer", DataTypes.IntegerType, true)
+                        .add("Context1Id.define_string", DataTypes.StringType, true)
+                        // Decimal precision currently hardcoded in QNameToDataTypeConverter
+                        .add("Context1Id.define_decimal", DataTypes.createDecimalType(28, 8), true)
+                        .add("Context1Id.define_boolean", DataTypes.BooleanType, true)
+        );
+
+        validateOutput(
+                context2IdFile.toURI().toString(),
+                Arrays.asList(
+                        // Expected Instants derived from reading the input files and printing out
+                        // the datetime column using UTC
+                        RowFactory.create(0, LocalDate.of(2002, 9, 27), Instant.parse("2002-05-31T13:19:08.512Z")),
+                        RowFactory.create(1, LocalDate.of(2002, 3, 10), Instant.parse("2002-05-07T09:19:19.31Z")),
+                        RowFactory.create(2, LocalDate.of(2002, 5, 29), Instant.parse("2002-03-13T13:00:52.859Z")),
+                        RowFactory.create(3, null, null),
+                        RowFactory.create(4, LocalDate.of(2002, 5, 21), Instant.parse("2002-02-17T08:59:43.793Z"))
+                ),
+                new StructType()
+                        .add("id", DataTypes.IntegerType, false)
+                        .add("Context2Id.define_date", DataTypes.DateType, true)
+                        .add("Context2Id.define_datetime", DataTypes.TimestampType, true)
+        );
+    }
+    
+    private void validateOutput(String filename, List<Row> expectedRows, StructType schema) {
+        spark = initializeSession(Java8API.ENABLED);
+        Dataset<Row> actualDataFrame = spark.read().parquet(filename);
+        // Column names with a dot in them need escaped with backticks
+        List<String> columnList = Arrays.asList(actualDataFrame.columns()).stream()
+                .map(x -> "`" + x + "`")
+                .collect(Collectors.toList());
```

Comment:
Do you think we should change our default column name encoder to use a different delimiter than period? Perhaps a dash would be better? Maybe a vertical bar / pipe?

Reply:
No strong feelings here, especially since I thought we were shortly going to allow users to specify output column names. Although I suppose we still might want a  default behavior if they don't specify anything.

Think a swap  is still worth it?

Reply:
I think there is a place for a default name encoder even when we add the ability for users to specify their own column names. The default encoder will need to get smarter so that it handles duplicates properly and maybe we also add a configuration option for column name delimiter there as well? That would alleviate from us the responsibility of choosing a universally acceptable value.  I just know that if you hit a problem with the column names, then the next person to use our solution is almost certainly going to hit the same problem. There is a workaround, so it isn't urgent. I'm just trying to make our stuff as easy to use as possible.